src/components/search-bar/SearchBar.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Input } from '../Input';
+import { Button } from '../Button';
+import './SearchBar.css';
+
+export type EventHandler<T> = (event: React.ChangeEvent<T>) => void;
+
+export interface SearchBarProps {
+  searchValue: string;
+  onChange: EventHandler<HTMLInputElement>;
+  onSubmit: EventHandler<HTMLFormElement>;
+}
+
+export class SearchBar extends React.Component<SearchBarProps> {
+  isError: boolean = false;
+
+  constructor(props: SearchBarProps) {
+    super(props);
+  }
+
+  static generateError() {
+    throw new Error('Test Error');
+  }
+
+  handleClickErrorButton() {
+    this.isError = true;
+    this.forceUpdate();
+  }
+
+  render() {
+    if (this.isError) {
+      SearchBar.generateError();
+      this.isError = false;
+    }
+    return (
+      <section className="search-bar">
+        <form className="search-bar__form" onSubmit={this.props.onSubmit}>
+          <Input
+            className="search-bar__input"
+            placeholder="Search animals by name"
+            value={this.props.searchValue}
+            onChange={this.props.onChange}
+          ></Input>
+          <Button className="search-bar__button" type="submit">
+            Search
+          </Button>
+          <Button
+            className="search-bar__button"
+            onClick={this.handleClickErrorButton.bind(this)}
```

Comment:
first of all - try to avoid any function created in jsx, it will lead to performance and render issues. you should do this before jsx, because otherwise every render your `onClick` would be a completely new function, which will cause a re-render of your button. for this binding you can use a [constructor](https://legacy.reactjs.org/docs/faq-functions.html#bind-in-constructor-es2015), or you can just simply use [array functions](https://legacy.reactjs.org/docs/faq-functions.html#class-properties-es2022)